rocketpy/mathutils/function.py

```diff
@@ -165,20 +174,20 @@ def set_source(self, source):
             source = source.get_source()
         # Import CSV if source is a string or Path and convert values to ndarray
         if isinstance(source, (str, Path)):
-            # Read file and check for headers
-            with open(source, mode="r") as f:
-                first_line = f.readline()
-            # If headers are found...
-            if first_line[0] in ['"', "'"]:
-                # Headers available
-                first_line = first_line.replace('"', " ").replace("'", " ")
-                first_line = first_line.split(" , ")
-                self.set_inputs(first_line[0])
-                self.set_outputs(first_line[1:])
-                source = np.loadtxt(source, delimiter=",", skiprows=1, dtype=float)
-            # if headers are not found
-            else:
-                source = np.loadtxt(source, delimiter=",", dtype=float)
+            with open(source, "r") as file:
+                first_line = file.readline()
+                try:
+                    # Try to convert the first line to floats
+                    np.array([float(item) for item in first_line.split(",")])
+                    # If successful, no headers are present
+                    file.seek(0)  # Reset file read position
+                    source = np.loadtxt(file, delimiter=",", dtype=float)
+                except ValueError:
+                    # If an error occurs, headers are present
+                    source = np.genfromtxt(
+                        source, delimiter=",", dtype=float, skip_header=1
+                    )
```

Comment:
This opens the same file twice, which seems like a code smell to me. Can we refactor it avoiding at all costs reading the same file twice?

Reply:
Same as Bressan's comment.

Just out of curiosity, how long do you imagine that could take to read a really really long file using genfromtxt? 
I tested with a 130k lines file these days, and the time is not significant. That worth a try at least.